Sources/Kafka/Utilities/DispatchQueueTaskExecutor.swift

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-client open source project
+//
+// Copyright (c) 2024 Apple Inc. and the swift-kafka-client project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-client project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=6.0)
+import Dispatch
+
+final class DispatchQueueTaskExecutor: TaskExecutor {
+    let queue: DispatchQueue
+
+    init(_ queue: DispatchQueue) {
+        self.queue = queue
+    }
+
+    public func enqueue(_ _job: consuming ExecutorJob) {
+        let job = UnownedJob(_job)
+        queue.async {
+            job.runSynchronously(
+                on: self.asUnownedTaskExecutor()
```

Comment:
if you're able to use the 

```
public func runSynchronously(isolatedTo serialExecutor: UnownedSerialExecutor,
                             taskExecutor: UnownedTaskExecutor) {
```


here and pass the queue's `UnownedSerialExecutor` that would be preferable as it would AFAIR be more correct in tracking where this is isolated to.

If this is a pain because the queue does not conform to SerialExecutor on some platforms still... then perhaps conditionalize it to platforms where it is? Or leave as is and let me know and we'll chase fixing the conformance.

Reply:
Got you. Since we own the underlying queue here is owned by us nothing should be isolated to it.